compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -795,7 +795,7 @@ private BType resolveTypedescTypeDesc(BLangConstrainedType td, ResolverData data
         BTypedescType constrainedType = new BTypedescType(symTable.empty, null);
         BTypeSymbol typeSymbol = type.tsymbol;
         constrainedType.tsymbol = Symbols.createTypeSymbol(typeSymbol.tag, typeSymbol.flags, typeSymbol.name,
-                typeSymbol.originalName, typeSymbol.pkgID, constrainedType, typeSymbol.owner,
+                typeSymbol.originalName, data.env.enclPkg.symbol.pkgID, constrainedType, typeSymbol.owner,
```

Comment:
shall we create a local var 
```
SymbolEnv symEnv = data.env;
 ```
 replace it in other places, seems it used in several places, pls check other places as well.

Reply:
Fixed in c5ad5f9dc0d36e45baa2a28b1a138d3104c18d4a. Applied the change for other places as well.